### PR TITLE
Colorbar docstring reorg

### PR DESCRIPTION
**Pull request by anntzer**
### Merge main and "other" params docs in colorbar.make_axes.
`colorbar.make_axes` (and `make_axes_gridspec`) are low-level enough
that it's not worth the complexity in docstring interpolation to split
out "main" parameters and "other" parameters, when the docstring of
the more commonly used `plt.colorbar` smashes them together.  (Also,
`pad` is not really much more low-level than `shrink` or `aspect`...)

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/colorbar.py [hunk 1]
```diff
@@ -29,98 +29,95 @@
...
+
 panchor : (float, float), or *False*, optional
     The anchor point of the colorbar parent axes. If *False*, the parent
     axes' anchor will be unchanged.
     Defaults to (1.0, 0.5) if vertical; (0.5, 0.0) if horizontal.
 """
 
 _colormap_kw_doc = """
+extend : {'neither', 'both', 'min', 'max'}
+    If not 'neither', make pointed end(s) for out-of- range values.  These are
```
Comment (QuLogic) [inline lib/matplotlib/colorbar.py:70]:
```suggestion
    If not 'neither', make pointed end(s) for out-of-range values.  These are
```

File: lib/matplotlib/colorbar.py [hunk 2]
```diff
@@ -29,98 +29,95 @@
...
+    set for a given colormap using the colormap set_under and set_over methods.
+
+extendfrac : {*None*, 'auto', length, lengths}
+    If set to *None*, both the minimum and maximum triangular colorbar
+    extensions with have a length of 5% of the interior colorbar length (this
```
Comment (QuLogic) [inline lib/matplotlib/colorbar.py:75]:
```suggestion
    extensions will have a length of 5% of the interior colorbar length (this
```

File: lib/matplotlib/colorbar.py [hunk 3]
```diff
@@ -29,98 +29,95 @@
...
-                  Whether to draw lines at color boundaries.
-    *label*       str
-                  The label on the colorbar's long axis.
-    ============  ====================================================
+label : str
+    The label on the colorbar's long axis.
 
     The following will probably be useful only in the context of
     indexed colors (that is, when the mappable has norm=NoNorm()),
     or other unusual circumstances.
```
Comment (QuLogic) [inline lib/matplotlib/colorbar.py:113]: This was about `value`/`boundaries`, but now appears in the docs for `label`, which is not right

File: lib/matplotlib/colorbar.py (hunk 1 above)
Comment (jklymak) [inline lib/matplotlib/colorbar.py:70]: Maybe avoid an awkward double negative as well?

```suggestion
    Make pointed end(s) for out-of-range values (unless 'neither').  These are
```

File: lib/matplotlib/colorbar.py [hunk 4]
```diff
@@ -29,98 +29,95 @@
...
+    the lengths of the minimum and maximum colorbar extensions respectively as
+    a fraction of the interior colorbar length.
+
+extendrect : bool
+    If *False* the minimum and maximum colorbar extensions will be triangular
+    (the default).  If *True* the extensions will be rectangular.
+
+spacing : {'uniform', 'proportional'}
+    Uniform spacing gives each discrete color the same space; proportional
+    makes the space proportional to the data interval.
```
Comment (jklymak) [inline lib/matplotlib/colorbar.py:95]:
```suggestion
    For discrete colorbars (`.BoudnaryNorm` or contours), 'uniform' gives each color the same 
    space; 'proportional' makes the space proportional to the data interval.
```

File: lib/matplotlib/colorbar.py [hunk 5]
```diff
@@ -29,98 +29,95 @@
...
-                   colormapped to the corresponding value in values
-                   will be used.
-    ============   ===================================================
-
+boundaries, values : None or a sequence
+    Normally only useful for indexed colors (i.e. ``norm=NoNorm()``).
+    If unset, the colormap will be displayed on a 0-1 scale.
+    If sequences, *values* must have a length 1 less than *boundaries*.  For
+    each region delimited by adjacent entries in *boundaries*, the colormapped
+    to the corresponding value in values will be used.
```
Comment (jklymak) [inline lib/matplotlib/colorbar.py:120]:
```suggestion
    each region delimited by adjacent entries in *boundaries*, the color that
    is mapped to the corresponding value in *values* will be used.
```

**Comment by anntzer** [comment]
Thanks, took in most edits.

**Review by jklymak — APPROVED** [review]

**Pull request merged**